recidiviz/calculator/bq/bq_load.py

```
@@ -0,0 +1,136 @@
+
+    try:
+        bq_schema = [
+            bigquery.SchemaField(
+                field['name'], field['type'], field['mode'])
+            for field in export_config.TABLE_EXPORT_SCHEMA[table_name]
+        ]
+    except KeyError:
+        logging.exception(
+            'Unknown table name "%s". Is it listed in '
+            'export_config.TABLES_TO_EXPORT?', table_name)
+        return None
+
+    job_config = bigquery.LoadJobConfig(
+        schema=bq_schema,
+        source_format=bigquery.SourceFormat.CSV,
+        write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE
+    )
+
+    load_job = bq_utils.client().load_table_from_uri(
+        uri,
+        table_ref,
+        job_config=job_config
+    )
+
+    logging.info('Started load job %s for table %s.%s.%s',
+                 load_job.job_id,
+                 table_ref.project, table_ref.dataset_id, table_ref.table_id)
+
+    return load_job, table_ref
+
+
+def wait_for_table_load(
+        load_job: bigquery.job.LoadJob,
+        table_ref: bigquery.table.TableReference):
+    """Wait for a table LoadJob to finish, and log its status.
+
+    Args:
+        load_job: BigQuery LoadJob whose result to wait for.
+        table_ref: TableReference to retrieve final table status.
+    """
+    try:
+        # Wait for table load job to complete.
+        load_job.result()
+        logging.info('Load job %s for table %s.%s.%s completed successfully.',
+                     load_job.job_id,
+                     table_ref.project,
+                     table_ref.dataset_id,
+                     table_ref.table_id)
+
+        destination_table = bq_utils.client().get_table(table_ref)
+        logging.info('Loaded %d rows in table %s.%s.%s',
+                     destination_table.num_rows,
+                     destination_table.project,
+                     destination_table.dataset_id,
+                     destination_table.table_id)
+    except exceptions.NotFound:
```

Is this raised by both of the calls above or only one of them?

---

Are there other exceptions that could be raised, that we should be catching, logging, and maybe propagating?

---

This was the only one I encountered, but I can imagine some ValueErrors or others sneaking in if tables or dataset names are not formatted correctly. I can catch all exceptions and surface them if you prefer?

---

(I left it at NotFound because I figured it is okay for yet-unknown exceptions to fail loudly)